netbox/extras/models/customfields.py

```diff
@@ -524,24 +524,36 @@ def to_form_field(self, set_initial=True, enforce_required=True, enforce_visibil
         # Object
         elif self.type == CustomFieldTypeChoices.TYPE_OBJECT:
             model = self.related_object_type.model_class()
-            field_class = CSVModelChoiceField if for_csv_import else DynamicModelChoiceField
-            field = field_class(
-                queryset=model.objects.all(),
-                required=required,
-                initial=initial,
-                query_params=self.related_object_filter
-            )
+            if for_csv_import:
+                field = CSVModelChoiceField(
+                    queryset=model.objects.all(),
+                    required=required,
+                    initial=initial,
+                )
+            else:
+                field = DynamicModelChoiceField(
+                    queryset=model.objects.all(),
+                    required=required,
+                    initial=initial,
+                    query_params=self.related_object_filter
+                )
```

Comment:
We can avoid duplicating these lines by preparing the field class' keyword arguments separately. (This needs to be done for the multi-object field type as well.)

```suggestion
            kwargs = {
                'queryset': model.objects.all(),
                'required': required,
                'initial': initial,
            }
            if not for_csv_import:
                kwargs['query_params'] = self.related_object_filter
            field = CSVModelChoiceField(**kwargs)
```